src/main/java/org/candlepin/subscriptions/subscription/SubscriptionSyncController.java

```diff
@@ -343,6 +345,39 @@ public void deleteSubscription(String subscriptionId) {
     subscriptionRepository.deleteBySubscriptionId(subscriptionId);
   }
 
+  @Transactional
+  public String terminateSubscription(String subscriptionId, OffsetDateTime terminationDate) {
+    var subscription =
+        subscriptionRepository
+            .findActiveSubscription(subscriptionId)
+            .orElseThrow(EntityNotFoundException::new);
+
+    var offering =
+        offeringRepository
+            .findById(subscription.getSku())
+            .orElseThrow(EntityNotFoundException::new);
+
+    // Wait until after we are sure there's an offering for this subscription before setting the
+    // end date.  We want validation to occur before we start mutating data.
+    subscription.setEndDate(terminationDate);
+
+    OffsetDateTime now = OffsetDateTime.now();
+    // The calculation returns a whole number, representing the number of complete units
+    // between the two temporals. For example, the amount in hours between the times 11:30 and
+    // 12:29 will zero hours as it is one minute short of an hour.
+    var delta = ChronoUnit.HOURS.between(terminationDate, now);
+    var productTag = tagProfile.tagForOfferingProductName(offering.getProductName());
+    if (tagProfile.isProductPAYGEligible(productTag) && delta > 0) {
```

Comment:
I found that delta is negative if termination date is in the past. `Math.abs(delta)` helps